packages/venia-concept/src/index.js

```diff
@@ -39,14 +40,38 @@ const authLink = setContext((_, { headers }) => {
 });
 
 // https://www.apollographql.com/docs/link/links/error/
-const errorLink = onError(({ graphQLErrors, networkError }) => {
+const errorLink = onError(({ graphQLErrors, networkError, response }) => {
     if (graphQLErrors)
         graphQLErrors.forEach(({ message, locations, path }) =>
             console.log(
                 `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
             )
         );
     if (networkError) console.log(`[Network error]: ${networkError}`);
+
+    const { data, errors } = response;
+
+    errors.forEach(({ message, path }, index) => {
+        // It's within the GraphQL spec to receive data and errors, where errors are merely informational and not
+        // intended to block. Almost all existing components were not built with this in mind, so we build special
+        // handling of this error message so we can deal with it at the time we deem appropriate.
+        if (message === 'Some of the products are out of stock.') {
```

Comment:
This makes the third file that checks against the error text. I think it's probably time to make a mapping, probably in venia-ui or peregrine, that way we can easily keep track of these strings. It'll make it easier to update in the future too, if/when they change.

https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/store/actions/cart/asyncActions.js#L415-L426

https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/ProductFullDetail/productFullDetail.js#L80-L108

Reply:
I feel like @supernova-at did this in a PR, and we heard some arguments for it being unnecessary. I agree referencing `MagentoGraphQLErrorMap.get('OUT_OF_STOCK')` might look cleaner, so happy to implement if you refresh my memory and help decide which package it should go in 🙂

Reply:
Oh I just meant a literal constants file. As peregrine uses strings, and peregrine does not import from venia-ui, it should live in peregrine :)

```js
// peregrine/utils/GraphqlConstants.js
export const SOME_PRODUCTS_OUT_OF_STOCK = 'Some of the products are out of stock.';
```

Reply:
There's an [open architecture proposal](https://github.com/magento/architecture/pull/393) to try and make this stuff a bit less painful.

Would it be easier for y'all if you got back codes for various errors, so you could match on those instead of a string? That way the backend could make the message clearer later without breaking pwa-studio.